File: my-folder/0904-leaf-similar-trees/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def leafSimilar(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> bool:
        def inOrder(root, v):
            if root:
                inOrder(root.left, v)
                if not root.left and not root.right:
                    v.append(root.val)
                inOrder(root.right, v)

        v1, v2 = [], []
        inOrder(root1, v1)
        inOrder(root2, v2)

        return v1 == v2

File: my-folder/0904-leaf-similar-trees/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */
var leafSimilar = function(root1, root2) {
    let l1 = [];
    let l2 = [];

    let dfs = function(root, l){
        if(root) {
            dfs(root.left, l);
            if(!root.left && !root.right) {
                l.push(root.val);
            }
            dfs(root.right, l);
        }
    }
    dfs(root1, l1);
    dfs(root2, l2);

    return JSON.stringify(l1) === JSON.stringify(l2);
};
